docs/backend/guides/how-to/create-a-custom-linter.md

```diff
@@ -0,0 +1,137 @@
+# How to Create a Custom Go Linter
+
+Creating custom GO linters can be a great way to analyze your 
+project's source code and alert you to bugs, errors, 
+or other issues with your code.
+
+## Setting up your linter
+
+### File structure
+Start by setting up the files you'll need for your linter:
+```golang
+-cmd
+    -linter_name
+        -main.go
+
+-pkg
+    -linter_name
+        -linter.go
+        -linter_test.go
+```
+In the `cmd` folder create a folder for your linter called `<linter_name>` and add an empty file called `main.go`
+
+In `pkg` folder create another folder for your linter called `<linter_name>` and in it you will 
+have your `linter.go` and `linter_test.go`
+
+### Analyzer:
+In the `cmd` folder contains the the analysis driver for your linter. The file `main.go` will utilize a package called `singlechecker`,
+which defines the main function for an analysis driver and provides a tool to run a standalone analysis.
+
+The code in `main.go` is very simple:
```

Comment:
Very small nitpick, and not even something that necessarily needs to change here, but I like to avoid using words like "simple", because that type of word can be used in many places where some might not find it simple. I like Django's writing style section on this: [Django writing style](https://docs.djangoproject.com/en/3.2/internals/contributing/writing-documentation/#writing-style). Not that I always succeed in avoiding that type of language, but I like to try to.

But like I said, in this case it, I don't mind it too much. Commenting more to share that writing style thing for future docs.

Reply:
Thank you for sharing this resource. You made a very good point so I changed the text.